tcp/conn.go

```diff
@@ -0,0 +1,25 @@
+package tcp
+
+import (
+	"fmt"
+	"net"
+)
+
+var _ net.Conn = &Conn{}
+
+// Conn contains the TCP connection.
+type Conn struct {
+	net.Conn
+}
+
+// NewConn returns a new TCP connection.
+func NewConn(addr string) (*Conn, error) {
+	conn, err := net.Dial("tcp", addr)
+	if err != nil {
+		return nil, fmt.Errorf("failed to dial: %w", err)
+	}
+
+	return &Conn{
+		conn,
+	}, nil
```

Comment:
```suggestion
	return &Conn{Conn: conn}, nil
```

Avoid setting fields implicitly when declaring structs

Reply:
I have updated the implementation by dropping `Conn` and just returning a `net.Conn`. This way we can avoid implicit embedding _and_ delegating calls to an underlying field.

WDYT?

Reply:
Amazing 😄